romdisk/date.asm

```diff
@@ -12,26 +12,69 @@
         ; A static buffer that can be used by any command implementation
         EXTERN init_static_buffer
         DEFC STATIC_DATE_BUFFER = init_static_buffer
+        DEFC TIME_OFFSET_IN_STRUCT = 5
 
-        ; date main routine. 
-        ; Print the current date and time on screen
+        ; Date main routine, print the current date and time on screen.
+        ; With zero arguments, just print date.
+        ; With one argument, set date; with two arguments, set date and time.
+        ; Date and time format is:
+        ;       YYYY-MM-DD HH:MM:SS
         ; Parameters:
         ;       HL - ARGV
-        ;       BC - ARGC
+        ;       BC - ARGC, < 256
         ; Returns:
         ;       A - 0 on success
         PUBLIC date_main
 date_main:
-        ; Ignore parameters
+        ; Read the existing date whether we are showing or setting.
+        ; If we are setting just the date, we need to load current time
+        ; so that we don't reset it.
+        ld de, STATIC_DATE_BUFFER
         ; Get system date:
         ;   DE - Address of the date structure to fill. Must be at least DATE_STRUCT_SIZE bytes.
         ; Returns:
         ;   A - ERR_SUCCESS on success, error code else
-        ld de, STATIC_DATE_BUFFER
+        push hl
         GETDATE()
+        pop hl
+        or a
+        jp nz, date_error
+        dec c ; Discount command name
+        or c
```

Comment:
`dec` on 8-bit registers already updates the flag registers, so you can omit this `or`